refinery/core/tests.py

```diff
@@ -939,20 +939,35 @@ def test__prepare_annotated_nodes_calls_methods_in_proper_order(
             ]
         )
 
-    def test___get_output_connection_to_analysis_result_mapping(self):
+    def create_analysis_results(self, include_faulty_result=False):
         common_params = {
             "analysis_uuid": self.analysis.uuid,
             "file_store_uuid": self.node.file_uuid,
             "file_name": self.node_filename,
             "file_type": self.node.get_file_store_item().filetype
         }
         analysis_result_0 = AnalysisResult.objects.create(**common_params)
-        AnalysisResult.objects.create(**common_params)
+
+        if include_faulty_result:
+            # This analysis result has a filename that doesn't correspond to
+            #  any specified AnalysisNodeConnection Output filenames
+            AnalysisResult.objects.create(
+                **dict(common_params, file_name="Bad Filename")
+            )
+        else:
+            AnalysisResult.objects.create(**common_params)
```

Comment:
You call `AnalysisResult.objects.create` 3 times but only store the result 2 times... Is that correct?

Reply:
Yeah, I'm only worried about the Analysis results that have a filename that corresponds to a derived result. `analysis_result_0` & `analysis_result_1` in this case